### PR TITLE
Issue 409 vmm parser iv

### PR DESCRIPTION
**Pull request by mortenjc**
### Issue reference / description

The branch you merge from should already reference an event-formation-unit github ticket number. You can add a descriptive title, but if an issue is referenced, you don't have to.

## Checklist for submitter

- [ ] Check for conflict with integration test
- [ ] Unit tests pass

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel.

### CODE REVIEW

**Comment by mortenjc** [comment]
This is mostly about adding support for ring configuration based on json files.
Since this allows to check for some configuration errors I have added these to FreiaInstrument.
There is still no full integration into the event production.

File: src/modules/freia/test/ConfigTest.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  uint16_t VMMOffset;
+};
+
+
+// This table generated from the ICD and will be
+// compared to the calculated values
+std::vector<RingCfg> FullConfig {
+  {  0, 2,  0,  0},
+  {  1, 2,  2,  8},
+
```
Comment (amues) [inline src/modules/freia/test/ConfigTest.cpp:41]: Is there a reason for the blank line here?

**Review by amues — APPROVED** [review]
Looks good, feel free to merge it.

File: src/modules/freia/test/ConfigTest.cpp (hunk 1 above)
Comment (mortenjc) [reply src/modules/freia/test/ConfigTest.cpp]: Yes that one is super important ;-)

No, not really. Must have slipped in by mistake.

**Pull request merged**